gdnative-core/src/core_types/quat.rs

```diff
@@ -9,22 +12,248 @@ pub struct Quat {
     pub w: f32,
 }
 
+/// Helper methods for `Quat`.
+///
+/// See the official [`Godot documentation`](https://docs.godotengine.org/en/stable/classes/class_quat.html).
 impl Quat {
+    pub const IDENTITY: Self = Self::new(0.0, 0.0, 0.0, 1.0);
+
+    /// Constructs a quaternion defined by the given values.
     #[inline]
-    pub fn new(x: f32, y: f32, z: f32, w: f32) -> Self {
+    pub const fn new(x: f32, y: f32, z: f32, w: f32) -> Self {
         Self { x, y, z, w }
     }
 
+    /// Constructs a quaternion from the given [Basis]
+    #[inline]
+    pub fn from_basis(basis: &Basis) -> Self {
+        basis.to_quat()
+    }
+
+    /// Constructs a quaternion that will perform a rotation specified by Euler angles (in the YXZ
+    /// convention: when decomposing, first Z, then X, and Y last), given in the vector format as
+    /// (X angle, Y angle, Z angle).
+    #[inline]
+    pub fn from_euler(euler: Vector3) -> Self {
+        Self::gd(glam::Quat::from_rotation_ypr(euler.y, euler.x, euler.z))
+    }
+
+    /// Constructs a quaternion that will rotate around the given axis by the specified angle. The
+    /// axis must be a normalized vector.
+    #[inline]
+    pub fn from_axis_angle(axis: Vector3, angle: f32) -> Self {
+        debug_assert!(axis.is_normalized(), "Axis is not normalized");
+        Self::gd(glam::Quat::from_axis_angle(axis.glam().into(), angle))
+    }
+
+    /// Performs a cubic spherical interpolation between quaternions `pre_a`, this quaternion, `b`,
+    /// and `post_b`, by the given amount `t`.
+    #[inline]
+    pub fn cubic_slerp(self, b: Self, pre_a: Self, post_b: Self, t: f32) -> Self {
+        let t2 = (1.0 - t) * t * 2.0;
+        let sp = self.slerp(b, t);
+        let sq = pre_a.slerpni(post_b, t);
+        sp.slerpni(sq, t2)
+    }
+
+    /// Returns the dot product of two quaternions.
+    #[inline]
+    pub fn dot(self, b: Self) -> f32 {
+        self.glam().dot(b.glam())
+    }
+
+    /// Returns Euler angles (in the YXZ convention: when decomposing, first Z, then X, and Y last)
+    /// corresponding to the rotation represented by the unit quaternion. Returned vector contains
+    /// the rotation angles in the format (X angle, Y angle, Z angle).
+    #[inline]
+    pub fn get_euler(self) -> Vector3 {
```

Comment:
Might be named `to_euler()` to make it symmetric with `from_euler()`?

It would be named differently than in GDScript, but there you also have overloaded constructors, so there's no `from_euler()`.